server/controllers/RequestController.js

```diff
@@ -0,0 +1,351 @@
+
+    // delete request from db
+    db.query(
+      'DELETE FROM requests WHERE id = $1 AND owner = $2 RETURNING *',
+      [requestId, req.user.email],
+    )
+      .then((result) => {
+        if (result.rowCount < 1) {
+          return res.status(404).json({
+            statusCode: 404,
+            error: [],
+            message: 'You have no request with that id, please try another request id',
+          });
+        }
+
+        res.status(200).json({
+          statusCode: 200,
+          error: [],
+          message: 'The request was succesfully deleted',
+        });
+      })
+      .catch(() => res.status(500).json({
+        statusCode: 500,
+        error: 'Internal Server Error',
+        message: 'Oops! Another bug must have crawled into our systems, ' +
+        'but we are right on it! Please try your request later :\'(',
+      }));
+  }
+
+  static updateRequest(req, res) {
+    // extract request id and request data
+    const { requestId } = req.params;
+    const {
+      type, item, model, detail,
+    } = req.body;
+
+    // db query to update request only if the reques status isn't pending
+    db.query(
+      `UPDATE requests SET type = $1, item = $2, model = $3, detail = $4, updated_at = $5 
+      WHERE id = $6 and owner = $7 and status = $8 RETURNING *`,
+      [
+        type, item, model, detail, 'NOW()',
+        requestId, req.user.email, 'in-review',
+      ],
+    )
+      .then((result) => {
+        if (result.rowCount < 1) {
+          return res.status(404).json({
+            status: 200,
+            error: [],
+            message: `You have no unapproved request with id ${req.params.requestId}. ` +
+            'You cannot edit a request that has been approved.',
+          });
+        }
+
+        res.status(200).json({
+          status: 200,
+          error: [],
+          message: 'You have successfully updated the request',
+          request: {
+            id: result.rows[0].id,
+            type: result.rows[0].type,
+            item: result.rows[0].item,
+            model: result.rows[0].model,
+            detail: result.rows[0].detail,
+            createdAt: result.rows[0].created_at,
+            updatedAt: result.rows[0].updated_at,
+          },
+        });
+      })
+      .catch(() => res.status(500).json({
+        statusCode: 500,
+        error: 'Internal Server Error',
+        message: 'Oops! Another bug must have crawled into our systems, ' +
+        'but we are right on it! Please try your request later :\'(',
+      }));
+  }
+
+
+  // Admin
+  static getAllRequests(req, res) {
+    // query db for all requests, ordering by id
+    db.query('SELECT * FROM requests ORDER BY id ASC')
+      .then((result) => {
+        if (result.rows < 1) {
+          return res.status(200).json({
+            statusCode: 200,
+            error: [],
+            message: 'There are no requests in the system. Do we even have people using this app?',
```

Comment:
But, as much as I love your response messages, they can sometimes be distracting. So I'll suggest you make the message more professional since you are not building a social app.